src/bls12_381/fq.rs

```diff
@@ -1120,6 +1135,32 @@ impl Fq {
         (self.0).0[5] = r11;
         self.reduce();
     }
+
+    pub fn parity(&self) -> bool {
+        (self.0).0[0] % 1 == 1
```

Comment:
Parity in our cases is whether or not the bit is lexicographically larger than its negative. (This allows us to use the same concept of parity for Fq2 as we do for Fq, and it's consistent with our serialization.) I think replacing this with:

```rust
pub(crate) fn parity(&self) -> bool {
    let mut neg = *self;
    neg.negate();
    self > &neg
}
```

I would also change `get_point_from_x` so that it uses this `parity` function, so that we can be sure the concepts are identical in hashing and serialization.